.github/workflows/build.yml

```diff
@@ -0,0 +1,91 @@
+name: build
+
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+  pull_request:
+
+permissions:
+  contents: write
+  id-token: write
+  packages: write
+
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_CLI_EXPERIMENTAL: "enabled"
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: arduino/setup-task@v1
+        with:
+          version: 3.x
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: docker/setup-qemu-action@v1
+      - uses: docker/setup-buildx-action@v1
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.17'
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ./dist/*.deb
+            ./dist/*.rpm
+            ./dist/*.apk
+          key: ${{ runner.os }}-go-${{ hashFiles('**/*.go') }}-${{ hashFiles('go.sum') }}--${{ hashFiles('.goreleaser.yml') }}
+      - run: task ci
+      - uses: codecov/codecov-action@v2
+        with:
+          file: ./coverage.txt
+      - run: git diff
+      - name: Install GoReleaser
+        if: startsWith(github.ref, 'refs/tags/v') ||  github.ref == 'refs/heads/main'
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          install-only: true
+      - name: goreleaser-release
+        if: startsWith(github.ref, 'refs/tags/v') ||  github.ref == 'refs/heads/main'
```

Comment:
release, but if in the goreleaser task additional logic that if its a tag ref will actually release, otherwise will just run a snapshot (generate everything but don't publish)

Reply:
Alright, adding the `version` flag fixed things indeed, workflows run fine. So if I understand correctly, now, if I tag I should see the package pop up in Fury, correct?

Reply:
yep, thats the hope! and if something goes sideways on the tag (lets say env var didn't take or the like) - you can always kick the build by deleting the tag, then force pushing until get a working build state

Reply:
Thanks, will try this soon!